packages/roosterjs-content-model-plugins/lib/tableEdit/editors/features/TableMover.ts

```diff
@@ -129,3 +157,214 @@ function isTableTopVisible(editor: IEditor, rect: Rect | null, contentDiv?: Node
+    initValue?.tableRect.remove();
+
+    setTableMoverCursor(editor, false);
+
+    if (element == context.div) {
+        // Table mover was only clicked, select whole table
+        selectWholeTable(table);
+    } else {
+        // Check if table was dragged on itself or Check if element is not in editor
+        if (
+            table.contains(element as Node) ||
+            !editor.getDOMHelper().isNodeInEditor(element as Node)
+        ) {
+            editor.setDOMSelection(initValue?.initialSelection ?? null);
+            return cancelMove(context);
+        }
+
+        const finalRange = editor.getDOMSelection();
+        if (initValue && finalRange?.type == 'range') {
+            // Move table to new position
+            finalRange.range.collapse(true);
+            let insertionSuccess: boolean = false;
+
+            // Insert table into content model
+            editor.formatContentModel(
+                (model, context) => {
+                    const SPArray = getSelectedSegmentsAndParagraphs(model, false);
+                    if (
+                        SPArray.length == 1 &&
+                        SPArray[0][0].segmentType == 'SelectionMarker' &&
+                        SPArray[0][1] != null
+                    ) {
+                        const paragraph = SPArray[0][1];
+                        paragraph.segments.forEach(segment => {
+                            if (segment == SPArray[0][0] && initValue.cmTable) {
+                                const doc = createContentModelDocument();
+                                doc.blocks.push(initValue.cmTable);
+                                insertionSuccess = !!mergeModel(model, doc, context, {
+                                    mergeFormat: 'none',
+                                });
+                                context.skipUndoSnapshot = true;
+                                // Add selection marker to the first cell of the table
+                                const [finalTable] = getFirstSelectedTable(model);
+
+                                if (finalTable) {
+                                    setSelection(model);
+                                    const FirstCell = finalTable.rows[0].cells[0];
+
+                                    const markerParagraph = FirstCell?.blocks[0];
+                                    if (markerParagraph?.blockType == 'Paragraph') {
+                                        const marker = createSelectionMarker(model.format);
+
+                                        markerParagraph.segments.unshift(marker);
+                                        setParagraphNotImplicit(markerParagraph);
+                                        setSelection(FirstCell, marker);
+                                    }
+                                } else {
+                                    // Restore initial selection
+                                    editor.setDOMSelection(initValue.initialSelection);
```

Comment:
do not call setDOMSelection inside formatContentModel, it will be overwrite when write back